.github/workflows/ci-build.yaml

```diff
@@ -0,0 +1,48 @@
+name: CI
+on: [push, pull_request]
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Install graphviz
+        shell: bash -l {0}
+        run: |
+          sudo apt-get update && sudo apt-get install graphviz
+
+      - name: Setup Conda Environment
+        uses: goanpeca/setup-miniconda@v1
+        with:
+          miniconda-version: "latest"
+          python-version: "3.7"
+          environment-file: binder/environment.yml
+          activate-environment: dask-examples
+          auto-activate-base: false
+
+      - name: Install testing and docs dependencies
+        shell: bash -l {0}
+        run: |
+          conda install -q pathlib nbconvert nbformat jupyter_client ipykernel
+          pip install nbsphinx dask-sphinx-theme sphinx
+
+      - name: Build
+        shell: bash -l {0}
+        run: sphinx-build -M html . _build -v
+
+      - name: Install SSH Client
+        if: ${{ github.ref == 'refs/heads/master' && github.event_name != 'pull_request'}}
+        uses: webfactory/ssh-agent@v0.4.1
+        with:
+          ssh-private-key: ${{ secrets.DEPLOY_KEY }}
+
+      - name: Deploy
+        if: ${{ github.ref == 'refs/heads/master' && github.event_name != 'pull_request'}}
+        uses: JamesIves/github-pages-deploy-action@3.7.1
```

Comment:
It's not critical, but I wanted to point out [`peaceiris/actions-gh-pages`](https://github.com/peaceiris/actions-gh-pages) as another possible action we could use for deploying docs. It uses the built in `secrets.GITHUB_TOKEN` auth token and so doesn't require an additional deploy key got setting up an ssh client. That said, I'm happy to defer to you here

Reply:
Hmm interesting. The one I've used here also supports that, but seems to advise in the docs that this can break with some repo configs and that a deploy key is more robust.

Happy to switch to the `GITHUB_TOKEN` approach to see if it works.